writer.go

```diff
@@ -0,0 +1,73 @@
+package hclog
+
+import (
+	"bytes"
+	"io"
+	"os"
+)
+
+type writer struct {
+	b bytes.Buffer
+	w io.Writer
+}
+
+func newWriter(w io.Writer) *writer {
+	return &writer{w: w}
+}
+
+func (w *writer) Flush(level Level) (err error) {
+	if lw, ok := w.w.(LevelWriter); ok {
+		_, err = lw.LevelWrite(level, w.b.Bytes())
+	} else {
+		_, err = w.w.Write(w.b.Bytes())
+	}
+	w.b.Reset()
+	return err
+}
+
+func (w *writer) Write(p []byte) (int, error) {
+	return w.b.Write(p)
+}
+
+func (w *writer) WriteByte(c byte) error {
+	return w.b.WriteByte(c)
+}
+
+func (w *writer) WriteString(s string) (int, error) {
+	return w.b.WriteString(s)
+}
+
+// LevelWriter is the interface that wraps the LevelWrite method.
+type LevelWriter interface {
+	LevelWrite(level Level, p []byte) (n int, err error)
+}
+
+// LeveledWriter writes error messages to stderr and all other log
+// messages to stdout.
+type LeveledWriter struct {
```

Comment:
Because this type satisfies your direct need, probably best to call it something like `StdoutExceptError`. Or if that's a weird name, maybe have the New function look like:

```golang
func NewLevelWriter(m map[Level]io.Writer) *LeveledWriter {
    return &LeveldWriter{ default: os.Stderr, overrides: m }
}
```

and then LevelWrite like:

```golang
func (w *LeveledWriter) LevelWrite(level Level, p []byte) (int, error) {
  o := w.override[level]
  if o == nil {
    o = w.default
  }

  return o.Write(p)
}
```

Reply:
My intention was to add a "ready to use" `LevelWriter` implementation that fitted (IMO) the most obvious use case, expecting that people would implement their own custom types for any specific custom use cases.

But I can see how making this more generic adds value and makes this default implementation much more flexible and usable for probably most, if not all, use cases. So I'll update the PR accordingly 👍